x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SnapshotAction.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.ilm.Step.StepKey;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class SnapshotAction implements LifecycleAction {
+
+    public static final String NAME = "snapshot";
+    public static final ParseField POLICY_FIELD = new ParseField("policy");
+
+    private static final ConstructingObjectParser<SnapshotAction, Void> PARSER = new ConstructingObjectParser<>(NAME,
+        a -> new SnapshotAction((String) a[0]));
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), POLICY_FIELD);
+    }
+
+    private final String policy;
+
+    public static SnapshotAction parse(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public SnapshotAction(String policy) {
+        if (policy == null) {
```

Comment:
We make want to change this to `Strings.hasText(policy)` so users can't specify `" "` on accident

Reply:
Done